tweak logic for using cached SDK client in config reloader (#313)
cognitedata/python-extractor-utils: CHANGELOG.md

- `Fixed` for any bug fixes.
- `Security` in case of vulnerabilities.

## 7.1.1

### Fixed

>   * Enhancement of `7.0.5`: more use cases covered (to avoid repeatedly fetching a new token).

Author: ```suggestion
  * Enhancement of `7.0.5`: more use cases covered (to avoid repeatedly fetching a new token).
  * When using remote config, the full local `idp-authentication` is now injected (some fields were missing).
```

>   * Enhancement of `7.0.5`: also avoid repeatedly fetching a new token before first config is stored/'accepted'.

Author: ```suggestion
  * Enhancement of `7.0.5`: more use cases covered (to avoid repeatedly fetching a new token).
```
